aws/aws.go

```diff
@@ -519,43 +567,63 @@ func IsNukeable(resourceType string, resourceTypes []string) bool {
 	return false
 }
 
+func nukeAllResourcesInRegion(account *AwsAccountResources, region string, session *session.Session) error {
+	resourcesInRegion := account.Resources[region]
+
+	for _, resources := range resourcesInRegion.Resources {
+		length := len(resources.ResourceIdentifiers())
+
+		// Split api calls into batches
+		logging.Logger.Infof("Terminating %d resources in batches", length)
+		batches := split(resources.ResourceIdentifiers(), resources.MaxBatchSize())
+
+		for i := 0; i < len(batches); i++ {
+			batch := batches[i]
+			if err := resources.Nuke(session, batch); err != nil {
+				// TODO: Figure out actual error type
+				if strings.Contains(err.Error(), "RequestLimitExceeded") {
+					logging.Logger.Info("Request limit reached. Waiting 1 minute before making new requests")
+					time.Sleep(1 * time.Minute)
+					continue
+				}
+
+				return errors.WithStackTrace(err)
+			}
+
+			if i != len(batches)-1 {
+				logging.Logger.Info("Sleeping for 10 seconds before processing next batch...")
+				time.Sleep(10 * time.Second)
+			}
+		}
+	}
+
+	return nil
+}
+
 // NukeAllResources - Nukes all aws resources
 func NukeAllResources(account *AwsAccountResources, regions []string) error {
 	for _, region := range regions {
+		// region that will be used to create a session
+		sessionRegion := region
+		var err error
```

Comment:
NIT: this is not necessary, as it gets defined a few lines below with `session, err := ...`